solidity/dashboard/src/utils/token.utils.js

```diff
@@ -0,0 +1,86 @@
+import BigNumber from "bignumber.js"
+
+const metrics = [
+  { divisor: 1, symbol: "" },
+  { divisor: 1e3, symbol: "K" },
+  { divisor: 1e6, symbol: "M" },
+]
+
+export function displayAmount(amount, withCommaSeparator = true) {
+  if (amount) {
+    const readableFormat = toTokenUnit(amount)
+    return withCommaSeparator
+      ? readableFormat.toFormat(0, BigNumber.ROUND_DOWN)
+      : readableFormat.toString()
+  }
+  return 0
+}
+
+/**
+ * Convert a weis amount to token units.
+ * @param {*} amount amount in weis
+ *
+ * @return {BigNumber} amount in token units
+ */
+export const toTokenUnit = (amount) => {
+  if (!amount) {
+    return new BigNumber(0)
+  }
+  return new BigNumber(amount).div(new BigNumber(10).pow(new BigNumber(18)))
+}
+
+/**
+ * Convert a token unit amount to weis.
+ * @param {*} amount amount in token units
+ * @param {*} decimals decimals
+ *
+ * @return {BigNumber} amount in weis
+ */
+export function fromTokenUnit(amount, decimals = 18) {
+  amount = new BigNumber(amount)
+  return amount.times(new BigNumber(10).pow(new BigNumber(decimals)))
+}
+
+/**
+ * Returns a number with a metric suffix eg.:
+ * * 10000000 => { value: 10, suffix: 'M', formattedValue: '10M' }
+ * * 1000.4 => { value: 1.4, suffix: 'K', formattedValue: '10K'}
+ * * 10000.4 => { value: 10, suffix: 'K', formattedValue: '10K' }
```

Comment:
10.4K ?

Reply:
We want to display decimals only if a value is less than 10.